assets/js/base/context/providers/cart-checkout/payment-methods/payment-method-data-context.tsx

```diff
@@ -213,68 +140,24 @@ export const PaymentMethodDataProvider = ( {
 
 	const currentStatus = useMemo(
 		() => ( {
-			isPristine: paymentData.currentStatus === PRISTINE,
-			isStarted: paymentData.currentStatus === STARTED,
-			isProcessing: paymentData.currentStatus === PROCESSING,
-			isFinished: [ ERROR, FAILED, SUCCESS ].includes(
-				paymentData.currentStatus
-			),
-			hasError: paymentData.currentStatus === ERROR,
-			hasFailed: paymentData.currentStatus === FAILED,
-			isSuccessful: paymentData.currentStatus === SUCCESS,
+			isPristine: paymentData.currentStatus === STATUS.PRISTINE,
+			isStarted: paymentData.currentStatus === STATUS.STARTED,
+			isProcessing: paymentData.currentStatus === STATUS.PROCESSING,
+			isFinished: [
+				STATUS.ERROR,
+				STATUS.FAILED,
+				STATUS.SUCCESS,
+			].includes( paymentData.currentStatus ),
+			hasError: paymentData.currentStatus === STATUS.ERROR,
+			hasFailed: paymentData.currentStatus === STATUS.FAILED,
+			isSuccessful: paymentData.currentStatus === STATUS.SUCCESS,
+			isDoingExpressPayment:
```

Comment:
I added this to make it easier to track when we're making payment using an express method. If the status is not pristine, and the method is express, we know this is true. It's being used to block the express area.